examples/maze/maze.rs

```diff
@@ -90,3 +40,53 @@ impl DerefMut for Maze {
         &mut self.grid
     }
 }
+
+/// Generates a new random maze
+pub fn generate<R: Rng>(rng: &mut R) -> Maze {
```

Comment:
The function is generic and outside the struct so that the struct need not be generic on `Rng`.